src/frontend/js/utils/asyncForEach.ts

```diff
@@ -0,0 +1,10 @@
+export async function asyncForEach<T>(
```

Comment:
I think it could lead to some sort of confusion with `waitFor` which waits for an `expect` to become valid. Where in this case it isn't what we do. If we put a non valid `expect` in the given callback, it will fail the first time and never retry.

And this function can also be used outside of tests.

Reply:
Let's go for `resolveAll`